lib/ansible/plugins/action/synchronize.py

```diff
@@ -155,6 +155,11 @@ def run(self, tmp=None, task_vars=None):
         if task_vars is None:
             task_vars = dict()
 
+        # We make a copy of the args here because we may fail and be asked to
+        # retry. If that happens we don't want to pass the munged args through
+        # to our next invocation. Munged args are single use only.
+        self._task._tmp_args = self._task.args.copy()
```

Comment:
_tmp_args should be a local variable rather than an instance attribute.  But if we do that the unittests will need changing in a more intrusive way.  I think that's the right choice but I haven't looked at how to achieve that yet.